server/registry/internal/storage/errors.go

```diff
@@ -47,76 +48,82 @@ func AlreadyExists(err error) bool {
 
 // grpcErrorForDBError converts recognized database error codes to grpc error codes.
 func grpcErrorForDBError(ctx context.Context, err error) error {
+	// unwrap err, if wrapped
+	cause := errors.Cause(err)
+
 	// if this error already has a gRPC status code, just return it
-	if _, ok := status.FromError(err); ok {
-		return err
+	if _, ok := status.FromError(cause); ok {
+		return cause
 	}
-	if err == context.DeadlineExceeded {
-		return status.Error(codes.DeadlineExceeded, err.Error())
+
+	log.Infof(ctx, "DBError %+v", err) // log the stack trace
```

Comment:
This ensures that errors from each call site are uniquely logged?

Reply:
Yes, this ensures we have a stack trace of the source error in the server log before we translate it for the client. This also potentially raises questions... Do we want to log this stack trace here? Do we want to log it at info level? Should we filter which errors are logged? But this strategy can be tweaked later based on how useful we find it in practice and this seems like a simple place to start.

Reply:
I'm not concerned about oversharing there.  We might want to push the log level to debug (@seaneganx ?), but I'm fine to defer that to a future change.

Reply:
BTW, only the first few levels of that call stack are informative.

Reply:
Yeah, I agree about the stack... but do you want us to write something to trim the stack?

Reply:
This seems like something appropriate for debug level. There should already be some kind of error level log being produced, although with less information (just the error message itself I think).

If this is logged at debug level I don't think we need to worry about trimming the stack. Printing one extra debug log per failed request doesn't seem very noisy to me.

Reply:
👍